src/openrct2/actions/TrackPlaceAction.hpp

```diff
@@ -68,31 +92,31 @@ DEFINE_GAME_ACTION(TrackPlaceAction, GAME_COMMAND_PLACE_TRACK, GameActionResult)
         if (ride == nullptr)
         {
             log_warning("Invalid ride for track placement, rideIndex = %d", (int32_t)_rideIndex);
-            return std::make_unique<GameActionResult>(
-                GA_ERROR::INVALID_PARAMETERS, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE, STR_NONE);
+            return std::make_unique<TrackPlaceActionResult>(
+                GA_ERROR::INVALID_PARAMETERS, STR_NONE);
         }
         if (ride->type == RIDE_TYPE_NULL)
         {
             log_warning("Invalid ride type, rideIndex = %d", (int32_t)_rideIndex);
-            return std::make_unique<GameActionResult>(
-                GA_ERROR::INVALID_PARAMETERS, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE, STR_NONE);
+            return std::make_unique<TrackPlaceActionResult>(
+                GA_ERROR::INVALID_PARAMETERS, STR_NONE);
         }
         rct_ride_entry* rideEntry = get_ride_entry(ride->subtype);
         if (rideEntry == nullptr)
         {
             log_warning("Invalid ride subtype for track placement, rideIndex = %d", (int32_t)_rideIndex);
-            return std::make_unique<GameActionResult>(
-                GA_ERROR::INVALID_PARAMETERS, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE, STR_NONE);
+            return std::make_unique<TrackPlaceActionResult>(
+                GA_ERROR::INVALID_PARAMETERS, STR_NONE);
         }
 
         if (!direction_valid(_origin.direction))
         {
             log_warning("Invalid direction for track placement, direction = %d", _origin.direction);
-            return std::make_unique<GameActionResult>(
-                GA_ERROR::INVALID_PARAMETERS, STR_RIDE_CONSTRUCTION_CANT_CONSTRUCT_THIS_HERE, STR_NONE);
+            return std::make_unique<TrackPlaceActionResult>(
+                GA_ERROR::INVALID_PARAMETERS, STR_NONE);
         }
 
-        auto res = std::make_unique<GameActionResult>();
+        auto res = std::make_unique<TrackPlaceActionResult>();
```

Comment:
All of the references to `gTrackGroundFlags` in this function and the Execute should be changed to our new variable `res->GroundFlagCount`